test/e2e/soak_namespace.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	//"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func extinguish(c *client.Client, totalNS int, maxAllowedAfterDel int, maxSeconds int) {
+
+	var err error
+
+	for n := 0; n < totalNS; n += 1 {
+		_, err = createTestingNS(fmt.Sprintf("nslifetest-%v", n), c)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	//deletion completed in maxSeconds
+	deletionCompletedTimeout := time.After(time.Duration(maxSeconds) * time.Second)
+
+	//start deleting after 10 seconds.
+	deletionStart := time.After(time.Duration(10 * time.Second))
+	tick := time.Tick(2 * time.Second)
+T:
+	for {
+		select {
+		case <-deletionCompletedTimeout:
+			Failf("Didn't delete enough namespaces... timeout %v was reached.", deletionCompletedTimeout)
+			break T
+		case <-deletionStart:
+			nsList, err := c.Namespaces().List(labels.Everything(), fields.Everything())
+			Expect(err).NotTo(HaveOccurred())
+			for _, item := range nsList.Items {
+				if strings.Contains(item.Name, "nslifetest") {
+					print(item.Name)
+					if err := c.Namespaces().Delete(item.Name); err != nil {
+						Failf("Failed deleting error ::: --- %v ", err)
+					}
+				}
+				Logf("namespace : %v api call to delete is complete ", item)
+			}
+
+		case <-tick:
+			nsList, err := c.Namespaces().List(labels.Everything(), fields.Everything())
+			Expect(err).NotTo(HaveOccurred())
+			var cnt = 0
+			for _, item := range nsList.Items {
+				if strings.Contains(item.Name, "nslifetest") {
+					cnt++
+				}
+			}
+			Logf("currently remaining NS == %v", cnt)
+
+			//The test only passes if this is true.  otherwise
+			//we will eventually reach the timeout.
+			if cnt <= maxAllowedAfterDel {
+				break T
+			}
+		}
+	}
+}
+
+var _ = Describe("ns-soak", func() {
```

Comment:
"Namespaces" not "ns-soak" please.
Concatenating the Describe string and the It string should ideally form a valid sentence, e.g.
"Namespaces should successfully delete" or whatever.
I don't make the rules, it's so-called Behavior Driven Development, and I'm just trying to keep our tests somewhat consistent in that regard.  I hope you understand :-)

http://onsi.github.io/ginkgo/#structuring-your-specs